### PR TITLE
WIP: add Koala Data Readers

### PR DESCRIPTION
**Pull request by rklasen**

### CODE REVIEW

File: genEnvVarFile.py [hunk 1]
```diff
@@ -80,7 +80,7 @@ def genEnvs():
     # * --------------
     vars["LMDFIT_MACROPATH"] = ask(
         "LMDFIT_MACROPATH",
-        f"{vars['VMCWORKDIR']}/macro/detectors/lmd",
```
Comment (rklasen) [inline genEnvVarFile.py:83]: This is just a default suggestion, the user is asked to change if needed. let's revert to the original

File: python/runKoaTrack.py [hunk 2]
```diff
@@ -72,6 +72,8 @@
     not check_stage_success(pathname + f"/Koala_Track_{start_evt}.root")
     or force_level == 1
 ):
+    os.chdir(macropath)
+    os.chdir(macropath)
```
Comment (rklasen) [inline python/runKoaTrack.py:76]: why two times?

File: python/runKoaTrack.py [hunk 3]
```diff
@@ -103,6 +105,8 @@
     not check_stage_success(pathname + f"/Koala_comp_{start_evt}.root")
     or force_level == 1
 ):
+    os.chdir(macropath)
+    os.chdir(macropath)
```
Comment (rklasen) [inline python/runKoaTrack.py:109]: why two times?

File: python/runKoaTrack.py [hunk 4]
```diff
@@ -116,12 +120,38 @@
         shell=True,
     )
 
+    os.chdir(macropath)
+    os.chdir(macropath)
```
Comment (rklasen) [inline python/runKoaTrack.py:124]: why two times?

File: python/makeMultipleFileListBunches.py [hunk 5]
```diff
@@ -38,8 +38,8 @@ def getListOfDirectories(path):
 
 
 # TODO: Oh ffs, read that from scenario config, NOT hardcode!
-filename_prefix = "Lumi_TrksQA_"
-# filename_prefix = "Koala_comp_"
```
Comment (rklasen) [inline python/makeMultipleFileListBunches.py:42]: we shouldn't hardcode this at all, there needs to be a reference to the config here

File: python/createMultipleLmdData.py [hunk 6]
```diff
@@ -180,7 +180,7 @@
     job = Job(
         resource_request,
         f"{lmdScriptPath}/singularityJob.sh {lmdScriptPath}/createLumiFitData.sh",
-        "createLumiFitData",
```
Comment (rklasen) [inline python/createMultipleLmdData.py:183]: we shouldn't hardcode this at all, there needs to be a reference to the config here

File: python/createLumiFitData.sh [hunk 7]
```diff
@@ -25,18 +25,18 @@ fi
 
 # if {filelist_url} is empty, the createLmdFitData binary must be run
 if [ -z ${filelist_url} ]; then
-  echo ${LMDFIT_BUILD_PATH}/bin/createLmdFitData -m $pbeam -t $type -c ${config_path} -d ${input_path} -n ${numEv} -e ${elastic_cross_section}
```
Comment (rklasen) [inline python/createLumiFitData.sh:28]: it's probaly better to not change this file and instead copy it for KOALA

File: python/lumifit/simulation.py [hunk 8]
```diff
@@ -172,7 +172,7 @@ def create_simulation_and_reconstruction_job(
     dirname_full = dirname + "/" + dirname_filter_suffix
     pathname_full = lmdfit_data_dir + "/" + dirname_full
     # TODO: this path depends on KOALA/LUMI, so change it accroding to scenario type!
-    macropath_full = os.environ["VMCWORKDIR"] + "/macro/detectors/lmd"
```
Comment (rklasen) [inline python/lumifit/simulation.py:175]: we shouldn't hardcode this at all, there needs to be a reference to the config here

File: python/lumifit/reconstruction.py [hunk 9]
```diff
@@ -138,13 +138,13 @@ def create_reconstruction_job(
         resource_request = make_test_job_resource_request()
 
     application_command = (
-        f"{lmdScriptPath}/singularityJob.sh {lmdScriptPath}/runLmdReco.py"
```
Comment (rklasen) [inline python/lumifit/reconstruction.py:141]: we shouldn't hardcode this at all, there needs to be a reference to the config here

File: python/lumifit/reconstruction.py [hunk 10]
```diff
@@ -138,13 +138,13 @@ def create_reconstruction_job(
...
+        f"{lmdScriptPath}/singularityJob.sh {lmdScriptPath}/runKoaReco.py"
     )
 
     job = Job(
         resource_request,
         application_url=application_command,
-        name="lmd_reco_",
```
Comment (rklasen) [inline python/lumifit/reconstruction.py:147]: we shouldn't hardcode this at all, there needs to be a reference to the config here

File: python/runKoaTrack.py (hunk 2 above)
Comment (JeFrech) [reply python/runKoaTrack.py:76]: I'm not sure, it might be that it just a problem from copying the previous code

**Comment by rklasen** [comment]
Okay all done, merging

**Pull request merged**